sys/malloc.c

```diff
@@ -171,11 +170,22 @@ static mem_block_t *try_allocating_in_area(mem_arena_t *ma,
   return mb;
 }
 
+void kmalloc_init(malloc_pool_t *mp, uint32_t pages, uint32_t pages_limit) {
+  TAILQ_INIT(&mp->mp_arena);
+  mtx_init(&mp->mutex, MTX_RECURSE);
+  kmalloc_add_pages(mp, pages);
+  mp->pages_used = pages;
+  mp->pages_limit = pages_limit;
+}
+
 void *kmalloc(malloc_pool_t *mp, size_t size, uint16_t flags) {
   size_t size_aligned = align(size, MB_ALIGNMENT);
+  if (size == 0 || size_aligned == 0)
```

Comment:
Following is true: `size_aligned == 0` <=> `size == 0`, so checking them both is superfluous.